packages/core/admin/admin/src/content-manager/pages/ListSettingsView/components/EditFieldForm.js

```diff
@@ -37,6 +38,10 @@ const EditFieldForm = ({
 }) => {
   const { formatMessage } = useIntl();
 
+  if (isEmpty(fieldToEdit)) {
+    return null;
+  }
```

Comment:
Code wise, this looks good to me - just wondering: how/ why would `fieldToEdit` become falsy?

Reply:
Good question, I've modified the fix now.

It seems that `EditFieldForm` was mounted with `fieldToEdit` set to '' because of the order we were calling things in `handleCloseModal`. Toggling the modal before dispatching the action fixes the bug and feels like a better fix.

I'm still not 100% clear on why clicking the modal close button would not cause this bug while using the 'esc' key would

Reply:
> I'm still not 100% clear on why clicking the modal close button would not cause this bug while using the 'esc' key would

I think this needs to be understood before merging this otherwise we may not be tackling the real issue here.

Reply:
I agree. It looks like as if this would tackle a symptom of another bug (e.g. execution order).

Reply:
this new fix make sense to me:
by dispatching
```js
    dispatch({
      type: 'UNSET_FIELD_TO_EDIT',
    });
```
we are setting: 
```js
draftState.fieldToEdit = '';
```

that fires a re-render of `EditFieldForm` with `fieldToEdit` as empty string (as it should)
so I think it makes sense to close the modal before resetting `fieldToEdit`

wdyt, am I missing something?

Reply:
Thanks all for your help. I have modified this fix to calculate whether the modal is open from `reducer.fieldForm`. This is an empty object whenever there is not a field selected for editing and can therefore be used to determine whether the modal should be open.

wdyt @gu-stav @ronronscelestes ?